### PR TITLE
Sharing tests

### PR DESCRIPTION
**Pull request by jnfrmarks**
Code for sharing tests

### CODE REVIEW

**Comment by karlitschek** [comment]
@schiesbn @PVince81

**Comment by PVince81** [comment]
Rebased/squashed and split the commits into relevant pieces.

Note that the two last ones would normally belong to a separate branch but let's keep them for now.

**Comment by PVince81** [comment]
@jnfrmarks have a look at https://github.com/owncloud/smashbox/commit/f2ebe314e20c6e400a1ac6728a1f113d29c25874 and tell me if you still want to keep those (we could submit this upstream)

The two last commits (test_storm fixes and config template tweaks) might normally belong to a separate branch/PR, but let's keep them here to save time.

@jnfrmarks you'll need to `git fetch` then delete the branch `git branch -D sharing-tests` then check it out again. This is because I changed history and regular pulling would cause conflicts.

**Comment by jnfrmarks** [comment]
I'm fine with deleting the .php files.

**Comment by PVince81** [comment]
Ok, I discarded the commit and moved it to a separate branch: https://github.com/owncloud/smashbox/tree/server-tools-additions (in case we still want it for some reason)

File: python/smashbox/utilities/__init__.py [hunk 1]
```diff
@@ -291,4 +369,197 @@ def fatal_check(expr,message):
...
+
+    cmd = "grep -i \"stat failed\" %s/owncloud.log" % d
+    rtnCode = runcmd (cmd, ignore_exitcode=True, log_warning=False)
+    error_check(rtnCode > 0,"\"Stat Failed\" message found in server log file")
+
+####### API Calls ############
+
+def get_oc_api ():
+
+#    from smashbox.utilities import ownCloudWebDav
```
Comment (PVince81) [inline python/smashbox/utilities/__init__.py]: You might want to do a quick cleanup pass and remove those ?

**Comment by jnfrmarks** [comment]
@PVince81 

One final commit to clean up the code some - not 100% but much better

**Comment by PVince81** [comment]
Looks fine :+1:

**Comment by jnfrmarks** [comment]
@PVince81 

I can't seem to close this PR - there's no close button for me.  Can you close this?

**Comment by PVince81** [comment]
A merged PR is automatically closed, nothing to do :smile:

**Comment by jnfrmarks** [comment]
I can't delete the branch because there's an open PR but I can't find the PR

**Comment by PVince81** [comment]
You can try this locally (make sure to double check):
`git push --delete origin sharing-tests`

**Pull request merged**